docusaurus/docs/create-a-plugin/extend-a-plugin/add-logs-metrics-traces-for-backend-plugins.md

```diff
@@ -0,0 +1,421 @@
+    ctxLogger.Warning("Warning msg", "key", "value")
+    ctxLogger.Error("Error msg", "error", errors.New("An error occurred"))
+}
+```
+
+#### Include additional contextual information in logs
+
+If you want to propagate additional contextual key-value pairs to subsequent code/logic you can use the [log.WithContextualAttributes](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/log#WithContextualAttributes) function.
+
+**Example:**
+
+```go
+func (ds *Datasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+    ctxLogger := ds.logger.FromContext(ctx)
+    ctxLogger.Debug("Debug msg", "someID", 1)
+    ctxLogger.Info("Info msg", "queryType", "default")
+    ctxLogger.Warning("Warning msg", "key", "value")
+    ctxLogger.Error("Error msg", "error", errors.New("An error occurred"))
+
+    for _, q := range req.Queries {
+        childCtx = log.WithContextualAttributes(ctx, []any{"refID", q.RefID, "queryType", q.QueryType})
+        ds.handleQuery(childCtx, q)
+    }
+}
+```
+
+### Best practices
+
+- Start the log message with a capital letter; for example, `logger.Info("Hello world")` instead of `logger.Info("hello world")`.
+- The log message should be an identifier for the log entry, try to avoid parameterization; for example, `logger.Debug(fmt.Sprintf(“Something happened, got argument %d”, “arg”))`, in favor of key-value pairs for additional data; for example, `logger.Info(“Something happened”, “argument”, “arg”)`.
+- Prefer using camelCase style when naming log keys; for example, `remoteAddr` or `userID`, to be consistent with Go identifiers.
+- Use the key `error` when logging Go errors; for example, `logger.Error("Something failed", "error", errors.New("An error occurred")`.
+- Use a contextual logger whenever you have access to a `context.Context`.
+- Do not log sensitive information, such as data source credentials or IP addresses, or other personally identifiable information.
+
+#### Validate and sanitize input coming from user input
+
+If log messages or key-value pairs originate from user input they should be validated and sanitized. Be careful to not expose any sensitive information in log messages (secrets, credentials, and so on). It's especially easy to do by mistake when including a Go struct as a value.
+
+#### When to use which log level?
+
+- **Debug:** Informational messages of high frequency and/or less-important messages during normal operations.
+- **Info:** Informational messages of low frequency and/or important messages.
+- **Warning:** Should in normal cases not be needed. If used, it should be actionable.
+- **Error:** Error messages indicating some operation failed (with an error) and the program didn't have a way to handle the error.
+
+:::note
+
+Incoming requests of high frequency are normally more common for the `QueryData` endpoint, since--for example--the nature of a dashboard generates a request per panel or query.
```

Comment:
```suggestion
Incoming requests of high frequency are normally more common for the `QueryData` endpoint, since - for example - the nature of a dashboard generates a request per panel or query.
```